frontend/packages/console-shared/src/components/dynamic-form/fields.tsx

```diff
@@ -26,75 +30,144 @@ export const DescriptionField: React.FC<FieldProps> = ({ id, description }) =>
     </span>
   ) : null;
 
-const FieldWrapper: React.FC<FieldProps> = ({
+export const DescriptionField: React.FC<FieldProps> = ({ id, description }) => (
+  <Description id={id} description={description} />
+);
+
+export const FormField: React.FC<FormFieldProps> = ({
   children,
-  name,
-  idSchema,
+  id,
+  defaultLabel,
   required,
   schema,
   uiSchema,
-  registry,
 }) => {
-  const options = getUiOptions(uiSchema);
-  const title = (options?.title || schema?.title || name) as string;
-  const description = options?.description || schema?.description;
-  const DescriptionFieldComponent = _.get(registry, 'DescriptionField', DescriptionField);
-
+  const [showLabel, label] = useSchemaLabel(schema, uiSchema, defaultLabel || 'Value');
   return (
-    <FormField id={idSchema.$id} displayTitle title={title} required={required}>
-      <>
-        {children}
-        {description && (
-          <DescriptionFieldComponent description={description} id={`${idSchema.$id}_description`} />
-        )}
-      </>
-    </FormField>
+    <div id={`${id}_field`} className="form-group">
+      {showLabel && label && (
+        <label className={classnames('form-label', { 'co-required': required })} htmlFor={id}>
+          {label}
+        </label>
+      )}
+      {children}
+    </div>
   );
 };
 
-export const ResourceRequirementsField: React.FC<FieldProps> = (props) => {
-  const { formData = {}, idSchema, onChange } = props;
-  return (
-    <FieldSet {...props}>
-      <dl id={idSchema.$id} style={{ marginLeft: '15px' }}>
-        <dt>Limits</dt>
-        <dd>
-          <ResourceRequirements
-            cpu={formData?.limits?.cpu || ''}
-            memory={formData?.limits?.memory || ''}
-            storage={formData?.limits?.['ephemeral-storage'] || ''}
-            onChangeCPU={(cpu) => onChange(_.set(_.cloneDeep(formData), 'limits.cpu', cpu))}
-            onChangeMemory={(mem) => onChange(_.set(_.cloneDeep(formData), 'limits.memory', mem))}
-            onChangeStorage={(sto) =>
-              onChange(_.set(_.cloneDeep(formData), 'limits.ephemeral-storage', sto))
-            }
-            path={`${idSchema.$id}.limits`}
-          />
-        </dd>
-        <dt>Requests</dt>
-        <dd>
-          <ResourceRequirements
-            cpu={formData?.requests?.cpu || ''}
-            memory={formData?.requests?.memory || ''}
-            storage={formData?.requests?.['ephemeral-storage'] || ''}
-            onChangeCPU={(cpu) => onChange(_.set(_.cloneDeep(formData), 'requests.cpu', cpu))}
-            onChangeMemory={(mem) => onChange(_.set(_.cloneDeep(formData), 'requests.memory', mem))}
-            onChangeStorage={(sto) =>
-              onChange(_.set(_.cloneDeep(formData), 'requests.ephemeral-storage', sto))
-            }
-            path={`${idSchema.$id}.requests`}
-          />
-        </dd>
-      </dl>
-    </FieldSet>
+export const FieldSet: React.FC<FieldSetProps> = ({
```

Comment:
Move `FieldSet` component from templates file to prevent dependency cycle.

Reply:
Same as `FormField` component. Refined props so that they are more easily reused or calculated in field or template components.